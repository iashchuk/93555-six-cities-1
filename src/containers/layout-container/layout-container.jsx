import React from "react";
import { connect } from "react-redux";

import Layout from "../../components/layout/layout.jsx";
import { getData } from "../../redux/fetch/actions";
import { checkLoginAsync } from "../../redux/auth/actions";

export const LayoutContainer = (props) => {
  return <Layout {...props} />;
};

const mapStateToProps = ({ auth }) => {
  return {
    isAuthenticated: auth.isAuthenticated,
    user: auth.user
  };
};

const mapDispatchToProps = {
  checkLoginAsync,
  getData
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(LayoutContainer);
