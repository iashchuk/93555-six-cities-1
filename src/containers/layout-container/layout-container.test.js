import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { LayoutContainer } from "./layout-container.jsx";

configure({ adapter: new Adapter() });

const isAuthenticated = true;
const getDataAsync = jest.fn();
const checkLoginAsync = jest.fn();

describe(`LayoutContainer`, () => {
  it(`renders correctly`, () => {
    const component = shallow(
        <LayoutContainer
          isAuthenticated={isAuthenticated}
          getDataAsync={getDataAsync}
          checkLoginAsync={checkLoginAsync}
        />
    );
    expect(component).toMatchSnapshot();
  });
});
