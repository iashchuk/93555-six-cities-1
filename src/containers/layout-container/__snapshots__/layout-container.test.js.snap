// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutContainer renders correctly 1`] = `ShallowWrapper {}`;
